### PR TITLE
Registration flow 071218

### PR DESCRIPTION
**Pull request by rtluu**
PR for registration flow. No validations or errors for input fields yet.

### CODE REVIEW

File: src/modules/register/Register.jsx [hunk 1]
```diff
@@ -1,146 +1,478 @@
...
+
+					<div className={'step' + status.back} onClick={this.goBack} />
+					<div className={'step' + status.next} onClick={this.showNext} />
+					<button className={status.submit} type={'submit'}>
+						Done
+					</button>
+				</div>
+
+				<div className="page-numbers">
+					<span>{page.number}</span>
```
Comment (jamesmcnamara) [inline src/modules/register/Register.jsx]: `this.screens.indexOf(active) + 1`

File: src/modules/register/Register.jsx [hunk 2]
```diff
@@ -1,146 +1,478 @@
...
-		this.setState({
-			password: e.target.value
-		});
+		var passwordString = JSON.stringify(e.target.value),
+			passwordLength = passwordString.replace(/\"/g, '').length;
+		if (passwordLength > 0) {
+			this.setState({passwordFocus: true});
+		}
+		this.setState({password: e.target.value});
+	};
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/register/Register.jsx]: nice, very well explained!

[the remark this answers is not in the document]
Comment (jamesmcnamara) [reply src/modules/register/Register.jsx]: Thanks!

Actually if we remove the `focus` variables as described, it's just:
```js
onChange = field => ({target: { value }})  => {
  this.setState({
    [field]: value,
  })
}; 
```

File: src/modules/register/Register.jsx [hunk 3]
```diff
@@ -1,146 +1,478 @@
...
+
+	render() {
+		var status = {},
+			focus = {},
+			page = {};
```
[the remark this answers is not in the document]
Comment (rtluu) [reply src/modules/register/Register.jsx]: Just refactored here

**Pull request merged**